apps/datahub/src/app/home/search/search-filters/search-filters.component.ts

```diff
@@ -70,6 +76,26 @@ export class SearchFiltersComponent implements OnInit {
     this.searchFacade.setSpatialFilterEnabled(enabled)
   }
 
+  toggleMyRecordsFilter(enabled: boolean) {
+    this.fieldsService
+      .buildFiltersFromFieldValues({ owner: enabled ? this.userId : [] })
+      .subscribe((filters) => this.searchService.updateFilters(filters))
+  }
+
+  myRecordsFilterEnabled() {
```

Comment:
This shouldn't be a function but a simple observable on the component, like `myRecordsFiltersEnabled$`